kore/src/Kore/Parser/Lexer.x

```diff
@@ -0,0 +1,325 @@
+tok x = tok' (const x)
+
+{- | Lexer action to construct an identifier token with a particular TokenClass
+using the current text of the token as its Text argument.
+-}
+tok_ident :: (Text -> TokenClass) -> AlexInput -> Int -> Alex Token
+tok_ident x = tok' (\s -> x s)
+
+{- | Lexer action to construct a string token with the TokenString TokenClass.
+The string is unescaped prior to its Text argument being placed inside the
+token.
+-}
+tok_string :: AlexInput -> Int -> Alex Token
+tok_string AlexInput{alexPosn=p@(AlexPn fp _ line column), alexStr} len = do
+  let text = decodeUtf8 (ByteString.take len alexStr)
+      unescaped = unescape text in case unescaped of
+                                        Left str -> alexError fp line column str
+                                        Right t -> return $ Token p (TokenString t)
+   
+-- | Data type for Tokens. Contains filename, location info, and TokenClass
+data Token = Token AlexPosn TokenClass
+  deriving stock (Eq)
+  deriving stock (Show)
+
+{- | Get the Text argument of a Token whose TokenClass contains such an 
+argument
+-}
+getIdentName :: Token -> Text
+getIdentName (Token _ (TokenIdent t)) = t
+getIdentName (Token _ (TokenSetIdent t)) = t
+getIdentName _ = error "getIdentName can only be called on TokenIdent or TokenSetIdent"
+
+
+-- | Get the TokenClass of a Token
+getTokenClass :: Token -> TokenClass
+getTokenClass (Token _ cls) = cls
+
+{- | Data type for the raw lexical data of a Token. Essentially an enumeration
+specifying which token a particular Token represents. Additionally contains the
+semantic data associated with identifier and string tokens.
+-}
+data TokenClass
+  = TokenModule
+  | TokenEndModule
+  | TokenImport
+  | TokenSort
+  | TokenSymbol
+  | TokenWhere
+  | TokenAlias
+  | TokenAxiom
+  | TokenClaim
+  | TokenHookedSort
+  | TokenHookedSymbol
+  | TokenColonEqual
+  | TokenColon
+  | TokenLeftBrace
+  | TokenRightBrace
+  | TokenLeftBracket
+  | TokenRightBracket
+  | TokenLeftParen
+  | TokenRightParen
+  | TokenComma
+  | TokenTop
+  | TokenBottom
+  | TokenNot
+  | TokenAnd
+  | TokenOr
+  | TokenImplies
+  | TokenIff
+  | TokenExists
+  | TokenForall
+  | TokenMu
+  | TokenNu
+  | TokenCeil
+  | TokenFloor
+  | TokenEquals
+  | TokenIn
+  | TokenNext
+  | TokenRewrites
+  | TokenDomainValue
+  | TokenLeftAssoc
+  | TokenRightAssoc
+  | TokenIdent Text
+  | TokenSetIdent Text
+  | TokenString Text
+  | TokenEOF
+  deriving stock (Eq, Show)
+
+-- -----------------------------------------------------------------------------
+-- Alex wrapper code.
+--
+-- Taken and modified from public domain code that is part of Alex:
+-- https://github.com/haskell/alex/blob/3.2.6/templates/wrappers.hs
+--
+-- Unlike the code in LexerWrapper.hs, this code must live in the same module
+-- as the scanner itself, so it was moved from that file to this one.
+
+
+alexScanUser :: () -> AlexInput -> Int -> AlexReturn (AlexInput -> Int -> Alex Token)
```

Comment:
Should be commented that this function will be generated by Alex and the type signature is needed to fix the `()` type.